rs/rust_canisters/random_traffic_test/src/main.rs

```diff
@@ -23,68 +29,44 @@ thread_local! {
     static CONFIG: RefCell<Config> = RefCell::default();
     /// The maximum number of calls each heartbeat will attempt to make.
     static MAX_CALLS_PER_HEARTBEAT: Cell<u32> = Cell::default();
+    /// A hasher used to generate unique call tree IDs.
+    static HASHER: RefCell<DefaultHasher> = RefCell::default();
+    /// An index for each attempted call; starts at 0 and then increments with each call.
+    static CALL_INDEX: Cell<u32> = Cell::default();
     /// A collection of records; one record for each call. Keeps track of how each call went,
     /// whether it was rejected or not and how many bytes we received as a reply.
-    static RECORDS: RefCell<Vec<Record>> = RefCell::default();
-}
-
-/// Calls `f` on a candid type taken from `arg_data()`; replies the result of `f`.
-fn with_candid_encoding<T, F>(f: F)
-where
-    T: candid::CandidType + for<'a> serde::Deserialize<'a>,
-    F: FnOnce(T) -> T,
-{
-    let msg = match candid::Decode!(&api::arg_data()[..], T) {
-        Ok(item) => Ok(f(item)),
-        Err(_) => Err(()),
-    };
-
-    let msg = candid::Encode!(&msg).unwrap();
-    api::reply(&msg[..]);
-}
-
-/// Sets the test config; returns the current config.
-#[export_name = "canister_update set_config"]
-fn set_config() {
-    with_candid_encoding(|config: Config| CONFIG.replace(config));
-}
-
-/// Sets the requests per round to be sent each heart beat; returns the current value.
-#[export_name = "canister_update set_max_calls_per_heartbeat"]
-fn set_max_calls_per_heartbeat() {
-    with_candid_encoding(|max_calls_per_heartbeat: u32| {
-        MAX_CALLS_PER_HEARTBEAT.replace(max_calls_per_heartbeat)
-    });
-}
-
-/// Sets the reply weight; returns the current weight.
-#[export_name = "canister_update set_reply_weight"]
-fn set_reply_weight() {
-    with_candid_encoding(|reply_weight: u32| REPLY_WEIGHT.replace(reply_weight));
+    static RECORDS: RefCell<BTreeMap<u32, Record>> = RefCell::default();
+    /// A counter for synchronous rejections.
+    static SYNCHRONOUS_REJECTIONS_COUNT: Cell<u32> = Cell::default();
```

Comment:
Added this counter mostly for completions sake and also if you use a config where everything is synchronously rejected (I mean you shouldn't but you could), all the records will get removed and so it might appear that the canister isn't doing anything. But this counter will still get incremented so you can tell that something is indeed happening.